tests/test_daq_reader.py

```diff
@@ -0,0 +1,161 @@
+                       })
+        with self.assertRaises(ValueError):
+            st.make(self.run_id, 'raw_records')
+
+    # Setup of tests like getting data and removing it after the test
+    # ---------------------------------------------------------------
+    @classmethod
+    def setUpClass(cls) -> None:
+        st = strax.Context()
+        st.register(DummyDAQReader)
+        st.storage = [strax.DataDirectory('./daq_test_data')]
+        st.set_config({'daq_input_dir': cls.live_data_path})
+        cls.st = st
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        path_live = f'live_data/{cls.run_id}'
+        if os.path.exists(path_live):
+            shutil.rmtree(path_live)
+            print(f'rm {path_live}')
+
+    def setUp(self) -> None:
+        if not os.path.exists(self.live_data_path):
+            print(f'Fetch {self.live_data_path}')
+            self.download_test_data()
+        rd = self.get_metadata()
+        st = self.set_context_config(self.st, rd)
+        self.st = st
+        self.assertFalse(self.st.is_stored(self.run_id, 'raw_records'))
+
+    def tearDown(self) -> None:
+        data_path = self.st.storage[0].path
+        if os.path.exists(data_path):
+            shutil.rmtree(data_path)
+            print(f'rm {data_path}')
+
+    # Small utility functions needed in setup
+    # ---------------------------------------------------------------
+    def download_test_data(self):
+        download_test_data(self.data_file)
+        self.assertTrue(os.path.exists(self.live_data_path))
+
+    def get_metadata(self):
+        md = get_resource(self.rundoc_file, fmt='json')
+        # This is a flat dict but we need to have a datetime object,
+        # since this is only a test, let's just replace it with a
+        # placeholder
+        md['start'] = datetime.now()
+        return md
+
+    @staticmethod
+    def set_context_config(st, run_doc):
+        """Update context with fields needed by the DAQ reader"""
+        daq_config = run_doc['daq_config']
+        st.set_context_config(dict(forbid_creation_of=tuple()))
+        st.set_config({
+            'channel_map':
+                dict(
+                    # (Minimum channel, maximum channel)
+                    # Channels must be listed in a ascending order!
+                    tpc=(0, 1),
+                    nveto=(1, 2),
+                    aqmon=(ARTIFICIAL_DEADTIME_CHANNEL, ARTIFICIAL_DEADTIME_CHANNEL + 1),
+                )})
+        update_config = {
+            'readout_threads': daq_config['processing_threads'],
+            'record_length': daq_config['strax_fragment_payload_bytes'] // 2,
+            'max_digitizer_sampling_time': 10,
+            'run_start_time': run_doc['start'].replace(tzinfo=timezone.utc).timestamp(),
+            'daq_chunk_duration': int(daq_config['strax_chunk_length'] * 1e9),
+            'daq_overlap_chunk_duration': int(daq_config['strax_chunk_overlap'] * 1e9),

```

Comment:
For most of these fields we directly reference `daq_config[field]` with one exception where we do `daq_config.get(field)`. Is there a guarantee that the test rundoc (and any future replacements) includes all the referenced fields, or do we want to protect ourselves and use `daq_config.get(field, redax_default)`?

Reply:
Right, this doc shouldn't change, especially as we are referencing a permalink, even if we update the github doc, we would first have to change the permalink first before it could hurt the test.

So generally you are right, but since we would be manually changing the URL to the doc, we'd also know if we have to change some of the fields so I suggest to leave it like this